### PR TITLE
Unsend friend request

### PR DESCRIPTION
**Pull request by daliaghosn**
In the updated ContactScreen, the friend request functionality has been enhanced to provide clearer visual feedback and allow for easy retraction of pending requests:

**1. Dynamic Button States for Friend Requests:**
- When a friend request is sent, the "Send Friend Request" button immediately changes to a gray "Requested" button. This makes it clear to the user that the request is pending, awaiting a response and that if the user clicks on it again, the request is un-sent.

**2. Unsend Friend Request Feature:**
- The new unsend functionality allows the user to cancel a previously sent request by clicking the "Requested" button again. Upon doing so, the request is withdrawn, and the button returns to its original purple "Send Friend Request" state. This improvement provides the user with control over pending requests, ensuring a more intuitive and forgiving user experience.

**3. Integration with the ViewModel and Repository:**
- These UI changes are backed by the unsendFriendRequest() function in the UserViewModel. When the user clicks the gray "Requested" button, this function calls the repository to delete the friend request from the database. If successful, the internal _sentFriendRequests state is updated, removing the friend from the list of sent requests. This change in state automatically updates the UI through StateFlows, returning the button to its original purple color.

**4. Tests for the Unsend Functionality:**
- Test coverage has been added to verify the behavior of unsendFriendRequest() and for the contact screen UI.

### CODE REVIEW

File: app/src/androidTest/java/com/swent/suddenbump/ui/contacts/ContactTest.kt [hunk 1]
```diff
@@ -80,51 +77,83 @@ class ContactScreenTest {
...
+    verify {
+      userViewModel.unsendFriendRequest(
+          user = match { it.uid == currentUser.value.uid },
+          friend = match { it.uid == friend.value.uid },
+          onSuccess = any(),
+          onFailure = any())
+    }
+
+    // If you want to check UI updates, depending on how the flows are updated,
+    // you might need to wait for the recomposition or flow update.
```
Comment (ifaresse) [inline app/src/androidTest/java/com/swent/suddenbump/ui/contacts/ContactTest.kt:147]: Maybe delete this comment

**Review by ifaresse — CHANGES_REQUESTED** [review]
Please remove the unnecessary comment, then it should be good !

**Review by ifaresse — APPROVED** [review]
LGTM !

File: app/src/androidTest/java/com/swent/suddenbump/ui/contacts/ContactTest.kt (hunk 1 above)
Comment (daliaghosn) [reply app/src/androidTest/java/com/swent/suddenbump/ui/contacts/ContactTest.kt]: Thanks, it's done.

**Pull request merged**